src/components/sign-out-button.tsx

```diff
@@ -0,0 +1,17 @@
+import { auth, signOut } from "@/auth";
+
+import { Button } from "./ui/button";
+
+export async function SignOutButton() {
+  const session = await auth();
+  return session === null ? null : (
+    <form
+      action={async () => {
+        "use server";
+        await signOut();
+      }}
+    >
+      <Button type="submit">Sign Out</Button>
+    </form>
+  );
+}
```

Comment:
W headerze strony głównej usunąłbym button z logowaniem i rejestracją i zamieniłbym `SignOutButton` na coś w rodzaju `AuthButton`, button gdzie jak ktoś jest zalogowany to pokazuje "Wyloguj się" a w przeciwnym wypadku "Zaloguj się". Do rejestracji można byłoby się dostać przez button "Nie masz jeszcze konta? Zarejestruj się" na stronie logowania
```suggestion
import Link from "next/link";

import { auth, signOut } from "@/auth";

import { Button } from "./ui/button";

export async function AuthButton() {
  const session = await auth();
  return session === null ? (
    <Button asChild>
      <Link href="/auth/login">Zaloguj się</Link>
    </Button>
  ) : (
    <form
      action={async () => {
        "use server";
        await signOut();
      }}
    >
      <Button type="submit">Wyloguj się</Button>
    </form>
  );
}
```

Reply:
ciekawy pomysł